labman/db/process.py

```diff
@@ -697,17 +782,32 @@ def create(cls, user, plates, plate_ext_id, robot):
             TRN.add(sql, [process_id, robot.id])
             instance = cls(TRN.execute_fetchlast())
 
-            # Create the output plate
-            # Magic number 3 -> 384-well plate
-            plate = plate_module.Plate.create(
-                plate_ext_id, plate_module.PlateConfiguration(3))
-
-            # Compress the plates
-            for i, in_plate in enumerate(plates):
-                row_pad = int(np.floor(i / 2))
-                col_pad = i % 2
-
-                instance._compress_plate(plate, in_plate, row_pad, col_pad)
+        # get the input plates' layouts (to avoid repeated sql calls)
+        plate_layouts = [x.layout for x in plates]
+
+        # Create the output plate
+        # Magic number 3 -> 384-well plate
+        plate_config_384 = plate_module.PlateConfiguration(3)
+        plate = plate_module.Plate.create(plate_ext_id, plate_config_384)
+
+        # Compress the plates
+        position_generator = GDNAPlateCompressionProcess.\
+            get_interleaved_quarters_position_generator(
+                len(plates), plate_config_384.num_rows,
+                plate_config_384.num_columns)
+
+        for p in position_generator:  # each position on interleaved plate
+            input_layout = plate_layouts[p.input_plate_order_index]
+            input_well = input_layout[p.input_row_index][p.input_col_index]
+            # completely empty wells (represented as Nones) are ignored
+            if input_well is not None:
+                # note adding 1 to the row/col from p, as those are
```

Comment:
When I said that it sounded messy is that in this block we +1 and in the block below (another method) we -1 so it sounds like the dev needs to be really aware of when +/-1 vs. accounting for that somewhere else, automatically.

Reply:
@AmandaBirmingham, can you open an issue about this? I'm not sure in which situations we need to +1 and -1 so perhaps you could add an explanation about this.